File: pipeline/src/main/java/com/ibrahimatay/stage/SortMessagesStage.java
package com.ibrahimatay.stage;

import com.ibrahimatay.model.Message;
import com.ibrahimatay.model.Messages;

import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class SortMessagesStage implements Stage<Messages> {
    @Override
    public Messages execute(Messages input) {
        List<Message> messages = input.messages();
        Comparator<Message> compareById = Comparator.comparingLong(Message::id);

        Collections.sort(messages, compareById);

        return new Messages(messages);
    }
}

File: Csharp/DesignPatterns.Pipeline/Stage/SortMessagesStage.cs
using System.Collections;
using DesignPatterns.Pipeline.Model;

namespace DesignPatterns.Pipeline.Stage;

public class SortMessagesStage : IStage<Messages>
{
    public Messages Execute(Messages input)
    {
        List<Message> messages = input.MessageList;

        messages.Sort(new MessageComparer());

        return new Messages(messages);
    }
}

public class MessageComparer : IComparer<Message>
{
    public int Compare(Message x, Message y)
    {
        if (ReferenceEquals(x, y)) return 0;
        if (ReferenceEquals(null, y)) return 1;
        if (ReferenceEquals(null, x)) return -1;
        var idComparison = x.Id.CompareTo(y.Id);
        if (idComparison != 0) return idComparison;
        return string.Compare(x.message, y.message, StringComparison.Ordinal);
    }
}